cpp/src/parquet/encoding.cc

```diff
@@ -3071,8 +3061,14 @@ class DeltaByteArrayDecoder : public DecoderImpl,
     prefix_len_offset_ = 0;
     num_valid_values_ = num_prefix;
 
+    int bytes_left = decoder_->bytes_left();
+    int suffix_begins = len - bytes_left;
+    if (suffix_begins < 0) {
+      throw ParquetException(
+          "Wrong data in DeltaByteArrayDecoder: bytes_left more than length");
+    }
```

Comment:
If this happens, doesn't that mean we have already read past the buffer end (buffer overflow)? Seems like that should be handled in the lengths decoder.